source/develop/release-management/features/virt/dedicated-cpu.md

```diff
@@ -0,0 +1,211 @@
+    allocated on thread siblings and threads from single core can only be
+    assigned to this VM. If some threads remain unused these will be excluded
+    from use by any other VM.
+
+
+## Prerequisites
+
+There are no special prerequisites for this feature.
+
+
+## Security
+
+This feature is primarily designed for performance tuning and is not meant to
+be a security feature. It is possible for other VMs to share the dedicated CPUs
+for a short period of time when the VM is brought up (boot, after migration,
+restore from snapshots, etc.). This is not so much problem during boot because
+by the time OS is started the CPUs should be already evicted, but it could
+potentially provide a window of opportunity (for exploiting CPU
+vulnerabilities) after finished migration or when restoring VM from snapshot.
+
+
+## Limitations
+
+See the *Security* section.
+
+Currently, when a CPU is already dedicated to some VM, VDSM will not move the
+VM onto another CPU to make it free to another VM. This may be addressed in
+the future (see *Future Work* section), but for now this means:
+
+  * It may be impossible to start VMs with CPU pinning on that host because
+    the requested CPUs are already occupied. It is suggested to start VMs with
+    CPU pinning before any VMs with dedicated CPU policy.
+
+  * It may be impossible to start a VM with dedicated CPUs even though there
+    seems to be enough space in terms of number of used vs. free CPUs.
+
+
+## Changes to Engine and Vdsm
+
+The high level overview of the workflow in Engine and Vdsm is following:
+
+1.  When refreshing host capabilities Engine picks the information about which
+    CPU is dedicated to VDSM process. It will also read a detailed information
+    about CPU topology on the host.
+
+2.  On VM start/migration/etc. Engine consults the policy for dedicating CPUs
+    and schedules the VM on host with enough free CPUs. Engine determines the
+    mapping between physical and virtual CPUs.
+
+3.  Engine produces domain XML that contains properly populated `<vcpupin>`
+    element (similarly to CPU pinning) and stores the policy name in VM
+    metadata.
+
+4.  Vdsm starts the VM and analyzes which CPUs were dedicated. Vdsm updates the
+    pool of shared CPUs and re-configures VMs without any CPU pinning or
+    dedicated CPUs to run on the new set of CPUs from shared pool.
+
+5.  When the VM with CPU pinning or dedicated CPUs is being destroyed, Vdsm
+    updates the pool of shared CPUs. Vdsm then re-configures VMs without any
+    CPU pinning or dedicated CPUs to run on the new set of CPUs from shared
+    pool. Because we allow overprovisioning CPUs this needs to be done
+    immediately to better utilize the CPUs among VMs.
+
+Several changes to the internal API are required. To accomplish step 1. the
+verb `Host.getCapabilities` will be extended to provide keys:
+
+  * `vdsmCpuAffinity`: list of one or more CPU IDs to which VDSM process is
```

Comment:
This is called `vdsmToCpusAffinity` now. But it's OK to update this and the following info later, based on the actual implementation.